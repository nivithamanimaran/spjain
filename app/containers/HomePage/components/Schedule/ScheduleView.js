import {
  Box,
  Button,
  Grid,
  IconButton,
  makeStyles,
  Typography,
  useTheme,
} from '@material-ui/core';
import React from 'react';
import moment from 'moment';
import KenSnackBar from '../../../../components/KenSnackbar';
import NavigateBeforeOutlinedIcon from '@material-ui/icons/NavigateBeforeOutlined';
import NavigateNextOutlinedIcon from '@material-ui/icons/NavigateNextOutlined';
import TUICalendar from '@toast-ui/react-calendar';
import AccessTimeIcon from '@material-ui/icons/AccessTime';
import { Link } from 'react-router-dom';
import LibraryBooksIcon from '@material-ui/icons/LibraryBooks';
import {
  KEY_USER_TYPE,
  KEY_DATE_FORMAT,
  KEY_STATUS,
} from '../../../../utils/constants';
import { useTranslation } from 'react-i18next';
import tuiTheme from './tuiTheme';
import Routes from '../../../../utils/routes.json';

const useStyles = makeStyles(theme => ({
  scheduler: {
    maxHeight: '40vh',
    overflowY: 'auto',
    overflowX: 'hidden',
    '&::-webkit-scrollbar': {
      width: '4px',
    },
    '&::-webkit-scrollbar-track': {
      opacity: '0.2',
    },
    '&::-webkit-scrollbar-thumb': {
      backgroundColor: `#787878`,
      opacity: '0.1',
      outline: `1px solid #787878`,
    },
  },
  iconButtons: {
    padding: 0,
  },
  headerItem: {
    flexGrow: 1,
  },
  textCenter: {
    textAlign: 'center',
  },
  popDetailsIcon: {
    width: 40,
    display: 'flex',
    alignItems: 'center',
  },
  link: {
    textDecoration: 'none',
    margintop: 10,
  },
  popDetailsItem: {
    padding: '5px 0',
    alignItems: 'center',
  },
  popup: {
    position: 'absolute',
    zIndex: 2000,
    background: 'white',
    borderRadius: 5,
    maxWidth: 350,
    maxHeight: 350,
    right: 0,
    boxShadow:
      '0px 8px 5px rgba(23, 43, 77, 0.04), 0px 15px 12px rgba(23, 43, 77, 0.08)',
    [theme.breakpoints.only('xs')]: {
      left: '0 !important',
      maxWidth: 'none',
      margin: 'auto',
      height: '50vh',
    },
  },
  cardHeader: {
    fontWeight: 600,
    color: theme.palette.KenColors.neutral100,
    textTransform: "uppercase",
    [theme.breakpoints.only('sm')]: {
      fontSize: '13px !important',
    },
  },

  scheduleDay: {
    padding: '6px !important',
    [theme.breakpoints.only('sm')]: {
      padding: '3px 2px !important',
    },
  },
  scheduleLabel: {
    textAlign: 'end',
  },

  schedule: {
    color: 'yellow',
  },
  someClass: {},
  lessonLink: {
    color: theme.palette.KenColors.primary,
    fontSize: '16px',
  },
  title: { fontSize: 14, fontWeight: 600, fontFamily: "Work Sans" },
}));

const ScheduleView = props => {
  const {
    user,
    snackbarData,
    handleSnackbarClose,
    getCalendar,
    updateCalendarDate,
    title,
    currentDate,
    calRef,
    schedules,
    onClickSchedule,
    config,
    popupStyles,
    popupRef,
    currentSchedule,
    setCurrentSchedule,
    startClass,
    joinClass,
    bbbLink,
    activityUrl,
  } = props;
  const classes = useStyles();
  const { t } = useTranslation();
  const theme = useTheme();
  const calendars = [
    {
      name: 'Calendar',
      id: '1',
      color: '#ffffff',
      bgColor: `${theme.palette.KenColors.background}`,
      dragBgColor: '#00a9ff',
      borderColor: `${theme.palette.KenColors.primary}`,
    },
  ];
  return (
    <Box p={2} data-testid={'schedule-view'} style={{padding: '0px'}}>
      <div>
        <KenSnackBar
          message={snackbarData.snackbarMessage}
          severity={snackbarData.snackbarSeverity}
          autoHideDuration={2000}
          open={snackbarData.openSnackbar}
          handleSnackbarClose={handleSnackbarClose}
          position="Bottom-Right"
        />
        <Grid container spacing={2} alignItems="center">
          <Grid
            item
            alignItems="center"
            className={classes.headerItem}
            sm={5}
            md={5}
          >
            <Typography className={classes.title}>
              {user.Type === KEY_USER_TYPE.faculty
                ? t('headings:My_Schedule').toUpperCase()
                : t('headings:Schedule').toUpperCase()}
            </Typography>
          </Grid>
          <Grid item sm={7} md={7} className={classes.scheduleLabel}>
            <IconButton
              aria-label="previous"
              className={classes.iconButtons}
              onClick={e => {
                getCalendar().prev();
                updateCalendarDate();
              }}
            >
              <NavigateBeforeOutlinedIcon />
            </IconButton>

            <Button
              aria-label="today"
              onClick={e => {
                // change current date

                if (title !== 'TODAY') {
                  getCalendar().today();
                  getCalendar().scrollToNow();
                  updateCalendarDate();
                }
              }}
              className={classes.scheduleDay}
              data-testid="current-date-button"
            >
              {' '}
              {title === 'TODAY' ||
                title === 'TOMORROW' ||
                title === 'YESTERDAY' ? (
                <div>{title}</div>
              ) : (
                <div>{currentDate}</div>
              )}
            </Button>

            <IconButton
              aria-label="next"
              className={classes.iconButtons}
              onClick={e => {
                getCalendar().next();
                updateCalendarDate();
              }}
            >
              <NavigateNextOutlinedIcon />
            </IconButton>
          </Grid>
        </Grid>
      </div>

      {currentDate && (
        <Box pt={2} className={classes.scheduler}>
          <TUICalendar
            usageStatistics={false}
            ref={calRef}
            view={'day'}
            // useCreationPopup={true}
            isReadOnly={true}
            useDetailPopup={false}
            calendars={calendars}
            schedules={schedules}
            onClickSchedule={onClickSchedule}
            taskView={[]}
            theme={tuiTheme(theme)}
            scheduleView={['time']}
            template={{
              time: function (schedule) {
                return (
                  `<div style="color: ${theme.palette.KenColors.neutral100
                  }; font-size: 14px; padding: 8px;font-weight: bold; ">` +
                  schedule.title +
                  '</div>'
                );
              },
            }}
            week={{
              hourStart: config?.schedule?.hourStart || 0,
              hourEnd: config?.schedule?.hourEnd || 24,
            }}

          // onBeforeCreateSchedule={onBeforeCreateSchedule}
          // onBeforeDeleteSchedule={onBeforeDeleteSchedule}
          // onBeforeUpdateSchedule={onBeforeUpdateSchedule}
          />
          <Box
            style={{ ...popupStyles }}
            id="popup"
            className={classes.popup}
            ref={popupRef}
          >
            {currentSchedule && (
              <Box p={2} data-testid={'current-schedule'}>
                <Grid container className={classes.popDetailsItem}>
                  <Grid item className={classes.popDetailsIcon}>
                    <div
                      style={{
                        height: 15,
                        width: 15,
                        background: currentSchedule.bgColor,
                        borderRadius: 3,
                      }}
                    >
                      {' '}
                    </div>
                  </Grid>
                  <Grid item>
                    <Typography variant="body1">
                      {currentSchedule.raw.hed__Course__cName}
                    </Typography>
                  </Grid>
                </Grid>
                <Grid container className={classes.popDetailsItem}>
                  <Grid item className={classes.popDetailsIcon}>
                    <AccessTimeIcon color="primary" />
                  </Grid>
                  <Grid item>
                    <Typography variant="body2">
                      {moment(currentSchedule.start.toUTCString()).format('LT')}{' '}
                      - {moment(currentSchedule.end.toUTCString()).format('LT')}
                    </Typography>
                  </Grid>
                </Grid>
                <Grid container className={classes.popDetailsItem}>
                  <Grid item className={classes.popDetailsIcon} />
                  <Grid item>
                    <Typography>
                      {user && user.Type === KEY_USER_TYPE.faculty ? (
                        <Button
                          variant="contained"
                          target="blank"
                          color="primary"
                          onClick={startClass}
                        >
                          {t('labels:Start_Class')}
                        </Button>
                      ) : user && user.Type === KEY_USER_TYPE.student ? (
                        <Button
                          variant="contained"
                          target="blank"
                          color="primary"
                          onClick={joinClass}
                        >
                          {t('labels:Join_Class')}
                        </Button>
                      ) : null}
                    </Typography>
                  </Grid>
                </Grid>
                <Grid container className={classes.popDetailsItem}>
                  <Grid item className={classes.popDetailsIcon} />
                  <Grid item>
                    {user && user.Type === KEY_USER_TYPE.faculty ? (
                      <Link
                        to={{
                          pathname: `/${Routes.classroom}`,
                          state: {
                            courseId: currentSchedule.raw.CourseOfferingID,
                            date: currentDate,
                            currentSchedule: currentSchedule,
                            bbbLink: bbbLink,
                          },
                        }}
                        className={classes.link}
                        onClick={e => {
                          currentSchedule.raw.meeting_link = bbbLink;
                        }}
                      >
                        <Button variant="contained" color="primary">
                          {t('labels:Go_To_Classroom')}
                        </Button>
                      </Link>
                    ) : null}
                  </Grid>
                </Grid>
                {activityUrl && (
                  <Grid container className={classes.popDetailsItem}>
                    <Grid item className={classes.popDetailsIcon}>
                      <LibraryBooksIcon color="primary" />
                    </Grid>
                    <Grid item>
                      <a
                        href={activityUrl}
                        target="_blank"
                        className={classes.lessonLink}
                        onClick={() => setCurrentSchedule(null)}
                      >
                        {t('labels:Go_To_Activities')}
                      </a>
                    </Grid>
                  </Grid>
                )}
              </Box>
            )}
          </Box>
        </Box>
      )}
    </Box>
  );
};

export default ScheduleView;
