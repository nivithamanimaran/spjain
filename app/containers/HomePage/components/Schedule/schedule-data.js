export const data = [
  {
    CourseOfferingID: 'a041y0000015Fs2AAE',
    Id: 'a031y000004ikfYAAQ',
    Location: 'a0I1y000000IbnEEAS',
    Name: 'CSE - Course I - Offering A Term 1',
    ProgramId: 'a0F1y000000Q4vWEAS',
    ProgramName: '2020-BTECH - Computer Science Engineering',
    RRULE:
      'DTSTART:20200701T000000Z;RRULE:FREQ=WEEKLY;INTERVAL=1;BYDAY=MO,TU,TH,FR;UNTIL=20240331T000000Z',
    hed__Course__cId: 'a051y000000gSXXAA2',
    hed__Course__cName: 'CSE - Course I',
    hed__End_Date__c: '2024-03-31',
    hed__End_Time__c: '09:45:00.000Z',
    hed__Friday__c: true,
    hed__Monday__c: true,
    hed__Saturday__c: false,
    hed__Start_Date__c: '2020-07-01',
    hed__Start_Time__c: '09:00:00.000Z',
    hed__Sunday__c: false,
    hed__Term__c: '2020-CSE Term 1',
    hed__Thursday__c: true,
    hed__Tuesday__c: true,
    hed__Wednesday__c: false,
    id: 'a0J1y000000g5ZdEAI',
    meeting_link: '',
    name: 'CSH-000002',
  },
  {
    CourseOfferingID: 'a041y0000015Fs2AAE',
    Id: 'a031y000004ikfYAAQ',
    Location: 'a0I1y000000IbnEEAS',
    Name: 'CSE - Course I - Offering A Term 1',
    ProgramId: 'a0F1y000000Q4vWEAS',
    ProgramName: '2020-BTECH - Computer Science Engineering',
    RRULE:
      'DTSTART:20200701T000000Z;RRULE:FREQ=WEEKLY;INTERVAL=1;BYDAY=MO,TU,WE,TH,FR;UNTIL=20240331T000000Z',
    hed__Course__cId: 'a051y000000gSXXAA2',
    hed__Course__cName: 'CSE - Course I',
    hed__End_Date__c: '2024-03-31',
    hed__End_Time__c: '14:45:00.000Z',
    hed__Friday__c: true,
    hed__Monday__c: true,
    hed__Saturday__c: false,
    hed__Start_Date__c: '2020-07-01',
    hed__Start_Time__c: '14:00:00.000Z',
    hed__Sunday__c: false,
    hed__Term__c: '2020-CSE Term 1',
    hed__Thursday__c: true,
    hed__Tuesday__c: true,
    hed__Wednesday__c: true,
    id: 'a0J1y000000g5ZiEAI',
    meeting_link: '',
    name: 'CSH-000003',
  },
];

export const user = {
  AccountId: '0011y000007WYfDAAW',
  AccountName: 'SRM University, Sonepat',
  ContactId: '0031y000008DKg2AAG',
  Name: 'Sundar Mohan',
  Phone: '9830921098',
  Program_Enrollment: [
    {
      Academic_Program: 'BTECH - Computer Science Engineering',
      Id: 'a081y000000aY2TAAU',
      Name: 'PE-0148',
      Program_Plan: '2020-BTECH - Computer Science Engineering',
      Status: 'Current',
      department: 'Computer Science',
      departmentId: '0011y000007WodoAAC',
      hed__Account__c: '0011y000007WpMyAAK',
      hed__Program_Plan__c: 'a0F1y000000Q4vWEAS',
    },
  ],
  Type: 'Student',
  address: '',
  mail: 'student1@ken42.onmicrosoft.com',
};

export const schedules = [
  {
    body: '',
    calendarId: '1',
    category: 'time',
    end: '2021-10-19T09:45:00',
    isVisible: true,
    raw: {
      CourseOfferingID: 'a041y0000015Fs2AAE',
      Id: 'a031y000004ikfYAAQ',
      Location: 'a0I1y000000IbnEEAS',
      Name: 'CSE - Course I - Offering A Term 1',
      ProgramId: 'a0F1y000000Q4vWEAS',
      ProgramName: '2020-BTECH - Computer Science Engineering',
      RRULE:
        'DTSTART:20200701T000000Z;RRULE:FREQ=WEEKLY;INTERVAL=1;BYDAY=MO,TU,TH,FR;UNTIL=20240331T000000Z',
      hed__Course__cId: 'a051y000000gSXXAA2',
      hed__Course__cName: 'CSE - Course I',
      hed__End_Date__c: '2024-03-31',
      hed__End_Time__c: '09:45:00.000Z',
      hed__Friday__c: true,
      hed__Monday__c: true,
      hed__Saturday__c: false,
      hed__Start_Date__c: '2020-07-01',
      hed__Start_Time__c: '09:00:00.000Z',
      hed__Sunday__c: false,
      hed__Term__c: '2020-CSE Term 1',
      hed__Thursday__c: true,
      hed__Tuesday__c: true,
      hed__Wednesday__c: false,
      id: 'a0J1y000000g5ZdEAI',
      meeting_link: '',
      name: 'CSH-000002',
    },
    start: '2021-10-19T09:00:00',
    title: 'CSE - Course I',
  },
  {
    body: '',
    calendarId: '1',
    category: 'time',
    end: '2021-10-19T14:45:00',
    isVisible: true,
    raw: {
      CourseOfferingID: 'a041y0000015Fs2AAE',
      Id: 'a031y000004ikfYAAQ',
      Location: 'a0I1y000000IbnEEAS',
      Name: 'CSE - Course I - Offering A Term 1',
      ProgramId: 'a0F1y000000Q4vWEAS',
      ProgramName: '2020-BTECH - Computer Science Engineering',
      RRULE:
        'DTSTART:20200701T000000Z;RRULE:FREQ=WEEKLY;INTERVAL=1;BYDAY=MO,TU,TH,FR;UNTIL=20240331T000000Z',
      hed__Course__cId: 'a051y000000gSXXAA2',
      hed__Course__cName: 'CSE - Course I',
      hed__End_Date__c: '2024-03-31',
      hed__End_Time__c: '09:45:00.000Z',
      hed__Friday__c: true,
      hed__Monday__c: true,
      hed__Saturday__c: false,
      hed__Start_Date__c: '2020-07-01',
      hed__Start_Time__c: '09:00:00.000Z',
      hed__Sunday__c: false,
      hed__Term__c: '2020-CSE Term 1',
      hed__Thursday__c: true,
      hed__Tuesday__c: true,
      hed__Wednesday__c: false,
      id: 'a0J1y000000g5ZdEAI',
      meeting_link: '',
      name: 'CSH-000002',
    },
    start: '2021-10-19T14:00:00',
    title: 'CSE - Course I',
  },
];

export const currentSchedule = {
  attendees: [],
  bgColor: '#F4F6FC',
  body: '',
  borderColor: '#00218D',
  calendarId: '1',
  category: 'time',
  color: '#ffffff',
  comingDuration: 0,
  customStyle: '',
  dragBgColor: '#00a9ff',
  dueDateClass: '',
  end: new Date(),
  goingDuration: 0,
  id: '',
  isAllDay: false,
  isFocused: false,
  isPending: false,
  isPrivate: false,
  isReadOnly: false,
  isVisible: true,
  location: '',
  raw: {
    CourseOfferingID: 'a041y0000015Fs2AAE',
    Id: 'a031y000004ikfYAAQ',
    Location: 'a0I1y000000IbnEEAS',
    Name: 'CSE - Course I - Offering A Term 1',
    ProgramId: 'a0F1y000000Q4vWEAS',
    ProgramName: '2020-BTECH - Computer Science Engineering',
    RRULE:
      'DTSTART:20200701T000000Z;RRULE:FREQ=WEEKLY;INTERVAL=1;BYDAY=MO,TU,TH,FR;UNTIL=20240331T000000Z',
    hed__Course__cId: 'a051y000000gSXXAA2',
    hed__Course__cName: 'CSE - Course I',
    hed__End_Date__c: '2024-03-31',
    hed__End_Time__c: '09:45:00.000Z',
    hed__Friday__c: true,
    hed__Monday__c: true,
    hed__Saturday__c: false,
    hed__Start_Date__c: '2020-07-01',
    hed__Start_Time__c: '09:00:00.000Z',
    hed__Sunday__c: false,
    hed__Term__c: '2020-CSE Term 1',
    hed__Thursday__c: true,
    hed__Tuesday__c: true,
    hed__Wednesday__c: false,
    id: 'a0J1y000000g5ZdEAI',
    meeting_link: '',
    name: 'CSH-000002',
  },
  recurrenceRule: '',
  start: new Date(),
  state: '',
  title: 'CSE - Course I',
};
