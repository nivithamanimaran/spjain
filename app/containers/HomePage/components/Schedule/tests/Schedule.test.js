import React from 'react';
import { render, cleanup, waitForElement } from 'react-testing-library';
import ReactDOM from 'react-dom';
import ThemeProvider from '../../../../../utils/themeProvider';
import { BrowserRouter as Router } from 'react-router-dom';
import { currentSchedule, data, schedules, user } from '../schedule-data';
import ScheduleView from '../ScheduleView';
import moment from 'moment';
afterEach(() => {
  cleanup();
});
let props = {
  user: user,
  snackbarData: {},
  handleSnackbarClose: jest.fn(),
  getCalendar: jest.fn(),
  updateCalendarDate: jest.fn(),
  title: 'TODAY',
  currentDate: moment().format('YYYY-MM-DD'),
  calRef: {},
  schedules: schedules,
  onClickSchedule: jest.fn(),
  config: {},
  popupStyles: {},
  popupRef: {},
  currentSchedule: undefined,
  setCurrentSchedule: jest.fn(),
  startClass: jest.fn(),
  joinClass: jest.fn(),
  bbbLink: undefined,
  activityUrl: undefined,
};
describe('Schedule', () => {
  const div = document.createElement('div');
  it('should render and match the snapshot', () => {
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <ScheduleView {...props} />
        </Router>
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <ScheduleView {...props} />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('schedule-view')).toMatchSnapshot();
  });

  it('Verify curent date button', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <ScheduleView {...props} />
        </Router>
      </ThemeProvider>
    );
    expect(getByTestId('current-date-button')).toBeTruthy();
    expect(getByTestId('current-date-button').textContent.trim()).toBe('TODAY');
  });

  it('Verify schedule popup', () => {
    props = { ...props, currentSchedule: currentSchedule };
    const { getByTestId } = render(
      <ThemeProvider>
        <Router>
          <ScheduleView {...props} />
        </Router>
      </ThemeProvider>
    );

    expect(getByTestId('current-schedule')).toBeTruthy();
  });

  it('Component should render if schedules is null', () => {
    const div = document.createElement('div');
    props = { ...props, schedules: null };
    ReactDOM.render(
      <ThemeProvider>
        <Router>
          <ScheduleView {...props} />
        </Router>
      </ThemeProvider>,
      div
    );
  });
});
