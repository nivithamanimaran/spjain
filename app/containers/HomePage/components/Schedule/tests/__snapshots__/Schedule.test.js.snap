// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule should render and match the snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-20"
  data-testid="schedule-view"
>
  <div>
    <div
      class="makeStyles-root-17"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center"
    >
      <div
        class="MuiGrid-root makeStyles-headerItem-3 MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-sm-5 MuiGrid-grid-md-5"
      >
        <p
          class="MuiTypography-root makeStyles-title-15 MuiTypography-body1"
        >
          HEADINGS:SCHEDULE
        </p>
      </div>
      <div
        class="MuiGrid-root makeStyles-scheduleLabel-11 MuiGrid-item MuiGrid-grid-sm-7 MuiGrid-grid-md-7"
      >
        <button
          aria-label="previous"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButtons-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.61 7.41L14.2 6l-6 6 6 6 1.41-1.41L11.03 12l4.58-4.59z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="today"
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-scheduleDay-10"
          data-testid="current-date-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
             
            <div>
              TODAY
            </div>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="next"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-iconButtons-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10.02 6L8.61 7.41 13.19 12l-4.58 4.59L10.02 18l6-6-6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-21 makeStyles-scheduler-1"
  >
    <div
      style="height: 800px;"
    >
      <div
        class="tui-full-calendar-layout"
        style="background-color: white;"
      >
        <div
          class="tui-view-15 tui-full-calendar-week-container"
        >
          <div
            class="tui-full-calendar-dayname-layout"
          >
            <div
              class="tui-full-calendar-dayname-container tui-view-16"
              style="border-top: 0px solid #e5e5e5; border-bottom: 0px solid #e5e5e5; height: 0px; background-color: inherit; text-align: left;"
            />
          </div>
          <div
            class="tui-full-calendar-vlayout-area tui-view-17 tui-full-calendar-vlayout-container"
            style="height: 0px;"
          >
            <div
              class="tui-view-19"
              data-auto-height="true"
              data-panel-index="0"
              style="position: relative;"
            >
              <div
                class="tui-full-calendar-timegrid-container tui-view-20"
              />
              <div
                class="tui-full-calendar-timegrid-sticky-container"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-22 makeStyles-popup-8"
      id="popup"
    />
  </div>
</div>
`;
